tests/integration-tests/src/commonTest/kotlin/test/WatcherErrorHandlingTest.kt

```diff
@@ -0,0 +1,274 @@
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import testFixtureToUtf8
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertIs
+
+@OptIn(ApolloExperimental::class)
+class WatcherErrorHandlingTest {
+  private lateinit var mockServer: MockServer
+  private lateinit var apolloClient: ApolloClient
+  private lateinit var store: ApolloStore
+
+  private suspend fun setUp() {
+    store = ApolloStore(MemoryCacheFactory(), cacheKeyGenerator = IdCacheKeyGenerator)
+    mockServer = MockServer()
+    apolloClient = ApolloClient.Builder().serverUrl(mockServer.url()).store(store).build()
+  }
+
+  private suspend fun tearDown() {
+    mockServer.stop()
+  }
+
+  @Test
+  fun fetchIgnoreAllErrors() = runTest(before = { setUp() }, after = { tearDown() }) {
+    val channel = Channel<EpisodeHeroNameQuery.Data?>()
+    val jobs = mutableListOf<Job>()
+
+    jobs += launch {
+      apolloClient.query(EpisodeHeroNameQuery(Episode.EMPIRE))
+          .fetchPolicy(FetchPolicy.CacheFirst)
+          .watch()
+          .collect {
+            channel.send(it.data)
+          }
+    }
+
+    jobs += launch {
+      apolloClient.query(EpisodeHeroNameQuery(Episode.EMPIRE))
+          .fetchPolicy(FetchPolicy.CacheOnly)
+          .watch()
+          .collect {
+            channel.send(it.data)
+          }
+    }
+
+    jobs += launch {
+      apolloClient.query(EpisodeHeroNameQuery(Episode.EMPIRE))
+          .fetchPolicy(FetchPolicy.NetworkFirst)
+          .watch()
+          .collect {
+            channel.send(it.data)
+          }
+    }
+
+    jobs += launch {
+      apolloClient.query(EpisodeHeroNameQuery(Episode.EMPIRE))
+          .fetchPolicy(FetchPolicy.NetworkOnly)
+          .watch()
+          .collect {
+            channel.send(it.data)
+          }
+    }
+
+    channel.assertEmpty()
+    jobs.forEach { it.cancel() }
+  }
+
+  @Test
+  fun refetchIgnoreAllErrors() = runTest(before = { setUp() }, after = { tearDown() }) {
+    val channel = Channel<EpisodeHeroNameQuery.Data?>()
+
+    val query = EpisodeHeroNameQuery(Episode.EMPIRE)
+
+    // The first query should get a "R2-D2" name
+    val job = launch {
+      mockServer.enqueue(testFixtureToUtf8("EpisodeHeroNameResponseWithId.json"))
+      apolloClient.query(query)
+          .fetchPolicy(FetchPolicy.NetworkOnly)
+          .refetchPolicy(FetchPolicy.NetworkOnly)
+          .watch()
+          .collect {
+            channel.send(it.data)
+          }
+    }
+    assertEquals(channel.receiveOrTimeout()?.hero?.name, "R2-D2")
+
+    // Another newer call gets updated information with "Artoo"
+    // Due to .refetchPolicy(FetchPolicy.NetworkOnly), a subsequent call will be executed in watch()
+    // we didn't enqueue anything in the mockServer so a network exception is thrown, and ignored by default
```

Comment:
Same here I think? If you want a network exception, enqueue an Error response else I would expect the MockServer to throw?